src/internal/storage/chunk/gc.go

```diff
@@ -53,23 +61,40 @@ func (gc *GarbageCollector) RunOnce(ctx context.Context) (retErr error) {
 		return errors.EnsureStack(err)
 	}
 	defer errors.Close(&retErr, rows, "close rows")
+	eg, ctx := errgroup.WithContext(ctx)
+	if !gc.isConcurrent {
+		eg.SetLimit(1)
+	}
 	for rows.Next() {
 		var ent Entry
 		if err := rows.StructScan(&ent); err != nil {
 			return errors.EnsureStack(err)
 		}
-		fields := []log.Field{zap.Stringer("chunkID", ent.ChunkID), zap.Uint64("gen", ent.Gen)}
-		if !ent.Uploaded {
-			log.Info(ctx, "possibility for untracked chunk", fields...)
-		}
-		if err := gc.deleteOne(ctx, ent); err != nil {
-			return err
-		}
-		log.Info(ctx, "deleting object for chunk entry", fields...)
+		eg.Go(func() error {
+			return gc.deleteHelper(ctx, ent)
+		})
+	}
+
+	if err := eg.Wait(); err != nil {
+		return errors.EnsureStack(err)
 	}
+
 	return errors.EnsureStack(rows.Err())
 }
 
+func (gc *GarbageCollector) deleteHelper(ctx context.Context, ent Entry) error {
```

Comment:
I think a name like `deleteOneWithLogs()` is a little more informative/self-documenting.